test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpStatementTest.cs

```diff
@@ -137,6 +137,168 @@ public void CatchClause()
                                ));
         }
 
+        public static TheoryData ExceptionFilterData
+        {
+            get
+            {
+                var factory = SpanFactory.CreateCsHtml();
+
+                // document, expectedStatement
+                return new TheoryData<string, StatementBlock>
+                {
+                    {
+                        "@try { someMethod(); } catch(Exception) when (true) { handleIO(); }",
+                        new StatementBlock(
+                            factory.CodeTransition(),
+                            factory
+                                .Code("try { someMethod(); } catch(Exception) when (true) { handleIO(); }")
+                                .AsStatement())
+                    },
+                    {
+                        "@try { A(); } catch(Exception) when (true) { B(); } finally { C(); }",
+                        new StatementBlock(
+                            factory.CodeTransition(),
+                            factory
+                                .Code("try { A(); } catch(Exception) when (true) { B(); } finally { C(); }")
+                                .AsStatement()
+                                .Accepts(AcceptedCharacters.None))
+                    },
+                    {
+                        "@try { A(); } catch(Exception) when (true) { B(); } catch(IOException) when (false) { C(); }",
+                        new StatementBlock(
+                            factory.CodeTransition(),
+                            factory
+                                .Code("try { A(); } catch(Exception) when (true) { B(); } catch(IOException) " +
+                                    "when (false) { C(); }")
+                                .AsStatement())
+                    },
+                    {
+                        string.Format("@try{0}{{{0}   A();{0}}}{0}catch(Exception) when (true)", Environment.NewLine) +
+                        string.Format("{0}{{{0}    B();{0}}}{0}catch(IOException) when (false)", Environment.NewLine) +
+                        string.Format("{0}{{{0}    C();{0}}}", Environment.NewLine),
+                        new StatementBlock(
+                            factory.CodeTransition(),
+                            factory
+                                .Code(
+                                    string.Format("try{0}{{{0}   A();{0}}}{0}catch(Exception) ", Environment.NewLine) +
+                                    string.Format("when (true){0}{{{0}    B();{0}}}{0}", Environment.NewLine) +
+                                    string.Format("catch(IOException) when (false){0}{{{0}    ", Environment.NewLine) +
+                                    $"C();{Environment.NewLine}}}")
+                                .AsStatement())
+                    },
+                    {
+                        "@{try { someMethod(); } catch(Exception) when (true) { handleIO(); }}",
```

Comment:
minor: Add a comment here because it's easy to miss the outermost curly braces (and think this is yet-another `@try` case).